### ISSUE TITLE
fatal error: concurrent map writes - gitlab pipeline - when running various terragrunt commands

### ISSUE THREAD

**Issue by Tomasz-Kluczkowski** [gruntwork-io/terragrunt#3614]
## Describe the bug

Hello,

We recently upgraded terragrunt from `0.45.0` to `0.69.1` and started observing intermittent error `fatal error: concurrent map writes` in gitlab pipelines.

Rerunning the job fixes the issue in first attempt (at least so far, never had to re-run second time).

Fragment of stack trace (see below for full):
```
fatal error: concurrent map writes
goroutine 15 [running]:
github.com/gruntwork-io/terragrunt/options.(*TerragruntOptions).AppendReadFile(0xc0009d3c08, {0xc000c6a410, 0x4a}, {0xc00103eaf0, 0x5d})
	/home/circleci/project/options/options.go:751 +0x1ec
github.com/gruntwork-io/terragrunt/config.ParseTerragruntConfig(0xc000a9a000, {0xc000c6a410?, 0xc000013151?}, 0x0)
	/home/circleci/project/config/config_helpers.go:646 +0x199
github.com/gruntwork-io/terragrunt/config.createTerragruntEvalContext.readTerragruntConfigAsFuncImpl.func6({0xc000425220, 0x1, 0x21dc0e0?}, {{0xc000b47250?, 0xc000aebb70?}})
```

Various `terragrunt` commands will fail (observed with `validate`, `plan`, `apply`) intermittently.

## Steps To Reproduce

Run enough times? Not sure if this is potential source for it but definitely when running many jobs at same time it is exposed more (might be a confirmation bias though, since recently that's the kind of pipelines I am running) ?

- run a pipeline running same command many times in parallel for different terragrunt modules?

## Expected behavior

Terragrunt commands succeed.

## Nice to haves

- [ ] Terminal output

```
fatal error: concurrent map writes
goroutine 15 [running]:
github.com/gruntwork-io/terragrunt/options.(*TerragruntOptions).AppendReadFile(0xc0009d3c08, {0xc000c6a410, 0x4a}, {0xc00103eaf0, 0x5d})
	/home/circleci/project/options/options.go:751 +0x1ec
github.com/gruntwork-io/terragrunt/config.ParseTerragruntConfig(0xc000a9a000, {0xc000c6a410?, 0xc000013151?}, 0x0)
	/home/circleci/project/config/config_helpers.go:646 +0x199
github.com/gruntwork-io/terragrunt/config.createTerragruntEvalContext.readTerragruntConfigAsFuncImpl.func6({0xc000425220, 0x1, 0x21dc0e0?}, {{0xc000b47250?, 0xc000aebb70?}})
	/home/circleci/project/config/config_helpers.go:706 +0x105
github.com/zclconf/go-cty/cty/function.Function.Call({0x2e9dee0?}, {0xc000425220, 0x1, 0x1})
	/home/circleci/go/pkg/mod/github.com/zclconf/go-cty@v1.14.2/cty/function/function.go:340 +0x6ee
github.com/hashicorp/hcl/v2/hclsyntax.(*FunctionCallExpr).Value(0xc0011d8d20, 0xc001926888)
	/home/circleci/go/pkg/mod/github.com/hashicorp/hcl/v2@v2.23.0/hclsyntax/expression.go:528 +0x1acf
github.com/hashicorp/hcl/v2/hclsyntax.(*RelativeTraversalExpr).Value(0xc000b99e30, 0xc001926888)
	/home/circleci/go/pkg/mod/github.com/hashicorp/hcl/v2@v2.23.0/hclsyntax/expression.go:171 +0x[37](https://gitlab.com/<REMOVED>/-/jobs/8522364760#L37)
github.com/hashicorp/hcl/v2/hclsyntax.(*FunctionCallExpr).Value(0xc0011d8ff0, 0xc001926888)
	/home/circleci/go/pkg/mod/github.com/hashicorp/hcl/v2@v2.23.0/hclsyntax/expression.go:493 +0x1304
github.com/gruntwork-io/terragrunt/config/hclparse.(*Attribute).Value(0xc001924050, 0xc0011d8ff0?)
	/home/circleci/project/config/hclparse/attributes.go:77 +0x2c
github.com/gruntwork-io/terragrunt/config.attemptEvaluateLocals(0xc000a9a000, 0xc0003cd560, {0xc001924030, 0x2, 0x0?}, 0xc00190c4b0)
	/home/circleci/project/config/locals.go:131 +0x51a
github.com/gruntwork-io/terragrunt/config.EvaluateLocalsBlock(0xc000a9a000, 0xc0003cd560)
	/home/circleci/project/config/locals.go:58 +0x206
github.com/gruntwork-io/terragrunt/config.DecodeBaseBlocks(0xc000c84[38](https://gitlab.com/<REMOVED>/-/jobs/8522364760#L38)0, 0xc0003cd560, 0xc0011f8420)
	/home/circleci/project/config/config_partial.go:171 +0x3a9
github.com/gruntwork-io/terragrunt/config.PartialParseConfig(0xc000d03e00, 0xc0003cd560, 0xc0011f8420)
	/home/circleci/project/config/config_partial.go:331 +0xcc
github.com/gruntwork-io/terragrunt/config.TerragruntConfigFromPartialConfig(0xc000d03e00, 0xc0003cd560, 0xc0011f8420)
	/home/circleci/project/config/config_partial.go:286 +0x3be
github.com/gruntwork-io/terragrunt/config.PartialParseConfigFile(0xc000d03e00, {0xc00105f200, 0x5f}, 0xc0011f8420)
	/home/circleci/project/config/config_partial.go:264 +0x1bf
github.com/gruntwork-io/terragrunt/config.partialParseIncludedConfig(0xc0011e9590?, 0xc0011d72e8?)
	/home/circleci/project/config/config_partial.go:607 +0xac
github.com/gruntwork-io/terragrunt/config.handleInclude(0xc000d03e00, 0xc0009731e0, 0x1)
	/home/circleci/project/config/include.go:124 +0x185
github.com/gruntwork-io/terragrunt/config.PartialParseConfig(0xc000d03a80, 0xc0003cc420, 0x0)
	/home/circleci/project/config/config_partial.go:558 +0xec6
github.com/gruntwork-io/terragrunt/config.TerragruntConfigFromPartialConfig(0xc000d03a80, 0xc0003cc420, 0x0)
	/home/circleci/project/config/config_partial.go:286 +0x3be
github.com/gruntwork-io/terragrunt/config.PartialParseConfigFile(0xc000d03a80, {0xc00103eaf0, 0x6c}, 0x0)
	/home/circleci/project/config/config_partial.go:264 +0x1bf
github.com/gruntwork-io/terragrunt/config.cloneTerragruntOptionsForDependencyOutput(0xc000c84580, {0xc00103eaf0, 0x6c})
	/home/circleci/project/config/dependency.go:693 +0x2f0
github.com/gruntwork-io/terragrunt/config.getTerragruntOutputJSON(0xc000c84580, {0xc00103eaf0, 0x6c})
	/home/circleci/project/config/dependency.go:743 +0x3a
github.com/gruntwork-io/terragrunt/config.getOutputJSONWithCaching(0xc000c84580, {0xc00103eaf0, 0x6c})
	/home/circleci/project/config/dependency.go:616 +0x345
github.com/gruntwork-io/terragrunt/config.getTerragruntOutput(0xc000c84580, {{0xc0014d4f90, 0x16}, 0xc000416f68, {{{0x2e9dee0, 0xc000013151}}, {0x21dc0e0, 0xc0010a6e20}}, 0x0, 0xc000357860, ...})
	/home/circleci/project/config/dependency.go:550 +0x90
github.com/gruntwork-io/terragrunt/config.getTerragruntOutputIfAppliedElseConfiguredDefault(0xc000c84580, {{0xc0014d4f90, 0x16}, 0xc000416f68, {{{0x2e9dee0, 0xc000013151}}, {0x21dc0e0, 0xc0010a6e20}}, 0x0, 0xc000357860, ...})
	/home/circleci/project/config/dependency.go:474 +0x1d8
github.com/gruntwork-io/terragrunt/config.(*Dependency).setRenderedOutputs(0xc000b99960, 0x0?)
	/home/circleci/project/config/dependency.go:164 +0x19b
github.com/gruntwork-io/terragrunt/config.dependencyBlocksToCtyValue.func1()
	/home/circleci/project/config/dependency.go:414 +0x87
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/home/circleci/go/pkg/mod/golang.org/x/sync@v0.9.0/errgroup/errgroup.go:78 +0x50
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 67
	/home/circleci/go/pkg/mod/golang.org/x/sync@v0.9.0/errgroup/errgroup.go:75 +0x96
goroutine 1 [semacquire]:
sync.runtime_Semacquire(0xc000c280a8?)
	/usr/local/go/src/runtime/sema.go:71 +0x25
sync.(*WaitGroup).Wait(0x251e8e0?)
	/usr/local/go/src/sync/waitgroup.go:118 +0x48
golang.org/x/sync/errgroup.(*Group).Wait(0xc000[39](https://gitlab.com/<REMOVED>/-/jobs/8522364760#L39)1600)
	/home/circleci/go/pkg/mod/golang.org/x/sync@v0.9.0/errgroup/errgroup.go:56 +0x25
github.com/gruntwork-io/terragrunt/cli.runAction(0xc000391380, 0xc0009d2808, 0xc000b7e6f0)
	/home/circleci/project/cli/app.go:239 +0x271
github.com/gruntwork-io/terragrunt/cli.NewApp.WrapWithTelemetry.func3.1({0x2e9d238?, 0xc000c1a2a0?})
	/home/circleci/project/cli/app.go:183 +0x7f
github.com/gruntwork-io/terragrunt/telemetry.Time({0x2e9d238?, 0xc000c1a2a0?}, {0x43651e0?, 0x4153ab?}, 0x7fcbe5b95e18?, 0xc000c28078?)
	/home/circleci/project/telemetry/metrics.go:42 +0x293
github.com/gruntwork-io/terragrunt/telemetry.Telemetry.func1({0x2e9d238?, 0xc000c1a2a0?})
	/home/circleci/project/telemetry/telemetry.go:80 +0x2d
github.com/gruntwork-io/terragrunt/telemetry.Trace({0x2e9d238?, 0xc000c1a2a0?}, {0x43651e0?, 0x0?}, 0xf98547d116986fc8?, 0xc000c28078?)
	/home/circleci/project/telemetry/traces.go:38 +0x14f
github.com/gruntwork-io/terragrunt/telemetry.Telemetry({0x2e9d238?, 0xc000c1a2a0?}, 0x277a447?, {0x43651e0?, 0x2?}, 0x21dc0e0?, 0xc000b1f[40](https://gitlab.com/<REMOVED>/-/jobs/8522364760#L40)1?)
	/home/circleci/project/telemetry/telemetry.go:79 +0x65
github.com/gruntwork-io/terragrunt/cli.NewApp.WrapWithTelemetry.func3(0xc000391380, 0xc000b7e6f0)
	/home/circleci/project/cli/app.go:172 +0x2c5
github.com/gruntwork-io/terragrunt/pkg/cli.Command.WrapAction.func1(0x0?)
	/home/circleci/project/pkg/cli/command.go:251 +0x1b
github.com/gruntwork-io/terragrunt/pkg/cli.(*Command).Run(0xc000bf2d20, 0xc000391340, {0xc000c2ad90?, 0x1, 0x0?})
	/home/circleci/project/pkg/cli/command.go:147 +0x5ea
github.com/gruntwork-io/terragrunt/pkg/cli.(*Command).Run(0xc000c3c000, 0xc000390180, {0xc000c2a2c0?, 0x1, 0x1?})
	/home/circleci/project/pkg/cli/command.go:142 +0x59b
github.com/gruntwork-io/terragrunt/pkg/cli.(*App).RunContext.func1(0xc000390040)
	/home/circleci/project/pkg/cli/app.go:1[41](https://gitlab.com/<REMOVED>/-/jobs/8522364760#L41) +0x3c5
github.com/urfave/cli/v2.(*Command).Run(0xc000c2c160, 0xc000390040, {0xc000406ba0, 0x2, 0x2})
	/home/circleci/go/pkg/mod/github.com/urfave/cli/v2@v2.27.5/command.go:276 +0x7e2
github.com/urfave/cli/v2.(*App).RunContext(0xc000bc0800, {0x2e9d238, 0xc000c1a2a0}, {0xc000406ba0, 0x2, 0x2})
	/home/circleci/go/pkg/mod/github.com/urfave/cli/v2@v2.27.5/app.go:333 +0x5a5
github.com/gruntwork-io/terragrunt/pkg/cli.(*App).RunContext(0xc0008b7ad0, {0x2e9d238, 0xc000c1a2a0}, {0xc000050040?, 0xc0008b7ad0?, 0xc0002b5740?})
	/home/circleci/project/pkg/cli/app.go:144 +0x287
github.com/gruntwork-io/terragrunt/cli.(*App).RunContext(0xc000b7e710, {0x2e9d238?, 0xc0003d1dd0?}, {0xc000050040, 0x2, 0x2})
	/home/circleci/project/cli/app.go:137 +0x290
main.main()
	/home/circleci/project/main.go:29 +0x149
goroutine 38 [select]:
go.opencensus.io/stats/view.(*worker).start(0xc00086da00)
	/home/circleci/go/pkg/mod/go.opencensus.io@v0.24.0/stats/view/worker.go:292 +0x9f
created by go.opencensus.io/stats/view.init.0 in goroutine 1
	/home/circleci/go/pkg/mod/go.opencensus.io@v0.24.0/stats/view/worker.go:34 +0x8d
goroutine 65 [syscall]:
os/signal.signal_recv()
	/usr/local/go/src/runtime/sigqueue.go:152 +0x29
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:23 +0x13
created by os/signal.Notify.func1.1 in goroutine 1
	/usr/local/go/src/os/signal/signal.go:151 +0x1f
goroutine 66 [select]:
github.com/gruntwork-io/terragrunt/internal/os/signal.NotifierWithContext.func1()
	/home/circleci/project/internal/os/signal/signal.go:27 +0xb2
created by github.com/gruntwork-io/terragrunt/internal/os/signal.NotifierWithContext in goroutine 1
	/home/circleci/project/internal/os/signal/signal.go:25 +0xc7
goroutine 67 [semacquire]:
sync.runtime_Semacquire(0xc0014168a0?)
	/usr/local/go/src/runtime/sema.go:71 +0x25
sync.(*WaitGroup).Wait(0x26034a0?)
	/usr/local/go/src/sync/waitgroup.go:118 +0x48
golang.org/x/sync/errgroup.(*Group).Wait(0xc00037d400)
	/home/circleci/go/pkg/mod/golang.org/x/sync@v0.9.0/errgroup/errgroup.go:56 +0x25
github.com/gruntwork-io/terragrunt/config.dependencyBlocksToCtyValue(0xc000c84580, {0xc001485340, 0x4, 0x0?})
	/home/circleci/project/config/dependency.go:449 +0x20a
github.com/gruntwork-io/terragrunt/config.decodeAndRetrieveOutputs(0xc000c84580, 0xc00039[42](https://gitlab.com/<REMOVED>/-/jobs/8522364760#L42)40)
	/home/circleci/project/config/dependency.go:224 +0x3af
github.com/gruntwork-io/terragrunt/config.ParseConfig(0xc000c84000, 0xc000394240, 0x0)
	/home/circleci/project/config/config.go:864 +0x28f
github.com/gruntwork-io/terragrunt/config.ParseConfigFile.func1({0xc000dfc080?, 0x1c000d376a8?})
	/home/circleci/project/config/config.go:787 +0x3d8
github.com/gruntwork-io/terragrunt/telemetry.Time({0x2e9d8a8?, 0xc000c84000?}, {0x27aacca?, 0x2e9d238?}, 0xc000ab0300?, 0x21dca20?)
	/home/circleci/project/telemetry/metrics.go:42 +0x293
github.com/gruntwork-io/terragrunt/config.ParseConfigFile.Telemetry.func2({0x2e9d8a8?, 0xc000c84000?})
	/home/circleci/project/telemetry/telemetry.go:80 +0x2d
github.com/gruntwork-io/terragrunt/telemetry.Trace({0x2e9d8a8?, 0xc000c84000?}, {0x27aacca?, 0xb?}, 0x[43](https://gitlab.com/<REMOVED>/-/jobs/8522364760#L43)650e0?, 0xc000c85f80?)
	/home/circleci/project/telemetry/traces.go:38 +0x14f
github.com/gruntwork-io/terragrunt/telemetry.Telemetry(...)
	/home/circleci/project/telemetry/telemetry.go:79
github.com/gruntwork-io/terragrunt/config.ParseConfigFile(0xc000c84000, {0xc000c9e2a0, 0x6e}, 0x0)
	/home/circleci/project/config/config.go:7[44](https://gitlab.com/<REMOVED>/-/jobs/8522364760#L44) +0x2a5
github.com/gruntwork-io/terragrunt/config.ReadTerragruntConfig({0x2e9d238, 0xc000c1a2a0}, 0xc0009d2808, {0xc0002bf140, 0x3, 0x3})
	/home/circleci/project/config/config.go:734 +0x19b
github.com/gruntwork-io/terragrunt/cli/commands/terraform.runTerraform({0x2e9d238, 0xc000c1a2a0}, 0xc0009d2808, 0xc001729ed0)
	/home/circleci/project/cli/commands/terraform/action.go:101 +0x1b4
github.com/gruntwork-io/terragrunt/cli/commands/terraform.Run({0x2e9d238?, 0xc000c1a2a0?}, 0xc000391380?)
	/home/circleci/project/cli/commands/terraform/action.go:83 +0x52
github.com/gruntwork-io/terragrunt/cli.NewApp.NewCommand.Action.func4(0xc000391380)
	/home/circleci/project/cli/commands/terraform/command.go:47 +0xca
github.com/gruntwork-io/terragrunt/cli.runAction.func2()
	/home/circleci/project/cli/app.go:233 +0x47
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/home/circleci/go/pkg/mod/golang.org/x/sync@v0.9.0/errgroup/errgroup.go:78 +0x50
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 1
	/home/circleci/go/pkg/mod/golang.org/x/sync@v0.9.0/errgroup/errgroup.go:75 +0x96
goroutine 14 [runnable]:
github.com/gruntwork-io/terragrunt/util.CloneStringMap(...)
	/home/circleci/project/util/collections.go:152
github.com/gruntwork-io/terragrunt/options.(*TerragruntOptions).Clone(0xc000544008, {0xc000ba4870, 0x4a})
	/home/circleci/project/options/options.go:578 +0x285
github.com/gruntwork-io/terragrunt/config.ParseTerragruntConfig(0xc000b96300, {0xc000ba4870?, 0xc000013151?}, 0x0)
	/home/circleci/project/config/config_helpers.go:652 +0x1b1
github.com/gruntwork-io/terragrunt/config.createTerragruntEvalContext.readTerragruntConfigAsFuncImpl.func6({0xc00032b9a0, 0x1, 0x21dc0e0?}, {{0xc00092eb90?, 0xc001447b70?}})
	/home/circleci/project/config/config_helpers.go:706 +0x105
github.com/zclconf/go-cty/cty/function.Function.Call({0x2e9dee0?}, {0xc00032b9a0, 0x1, 0x1})
	/home/circleci/go/pkg/mod/github.com/zclconf/go-cty@v1.14.2/cty/function/function.go:340 +0x6ee
github.com/hashicorp/hcl/v2/hclsyntax.(*FunctionCallExpr).Value(0xc001256e10, 0xc001a02168)
	/home/circleci/go/pkg/mod/github.com/hashicorp/hcl/v2@v2.23.0/hclsyntax/expression.go:528 +0x1acf
github.com/hashicorp/hcl/v2/hclsyntax.(*RelativeTraversalExpr).Value(0xc0004ae460, 0xc001a02168)
	/home/circleci/go/pkg/mod/github.com/hashicorp/hcl/v2@v2.23.0/hclsyntax/expression.go:171 +0x37
github.com/hashicorp/hcl/v2/hclsyntax.(*FunctionCallExpr).Value(0xc0012570e0, 0xc001a02168)
	/home/circleci/go/pkg/mod/github.com/hashicorp/hcl/v2@v2.23.0/hclsyntax/expression.go:493 +0x1304
github.com/gruntwork-io/terragrunt/config/hclparse.(*Attribute).Value(0xc001215110, 0xc0012570e0?)
	/home/circleci/project/config/hclparse/attributes.go:77 +0x2c
github.com/gruntwork-io/terragrunt/config.attemptEvaluateLocals(0xc000b96300, 0xc00030cf00, {0xc001215100, 0x2, 0x0?}, 0xc0018947e0)
	/home/circleci/project/config/locals.go:131 +0x51a
github.com/gruntwork-io/terragrunt/config.EvaluateLocalsBlock(0xc000b96300, 0xc00030cf00)
	/home/circleci/project/config/locals.go:58 +0x206
github.com/gruntwork-io/terragrunt/config.DecodeBaseBlocks(0xc000b96680, 0xc00030cf00, 0xc001287b60)
	/home/circleci/project/config/config_partial.go:171 +0x3a9
github.com/gruntwork-io/terragrunt/config.PartialParseConfig(0xc000b96280, 0xc00030cf00, 0xc001287b60)
	/home/circleci/project/config/config_partial.go:331 +0xcc
github.com/gruntwork-io/terragrunt/config.TerragruntConfigFromPartialConfig(0xc000b96280, 0xc00030cf00, 0xc001287b60)
	/home/circleci/project/config/config_partial.go:286 +0x3be
github.com/gruntwork-io/terragrunt/config.PartialParseConfigFile(0xc000b96280, {0xc00121c7e0, 0x52}, 0xc001287b60)
	/home/circleci/project/config/config_partial.go:264 +0x1bf
github.com/gruntwork-io/terragrunt/config.partialParseIncludedConfig(0xc001286db0?, 0xc0012732e8?)
	/home/circleci/project/config/config_partial.go:607 +0xac
github.com/gruntwork-io/terragrunt/config.handleInclude(0xc000b96280, 0xc000c2c000, 0x1)
	/home/circleci/project/config/include.go:124 +0x185
github.com/gruntwork-io/terragrunt/config.PartialParseConfig(0xc0003f5780, 0xc00030c080, 0x0)
	/home/circleci/project/config/config_partial.go:558 +0xec6
github.com/gruntwork-io/terragrunt/config.TerragruntConfigFromPartialConfig(0xc0003f5780, 0xc00030c080, 0x0)
	/home/circleci/project/config/config_partial.go:286 +0x3be
github.com/gruntwork-io/terragrunt/config.PartialParseConfigFile(0xc0003f5780, {0xc00121c0c0, 0x5f}, 0x0)
	/home/circleci/project/config/config_partial.go:264 +0x1bf
github.com/gruntwork-io/terragrunt/config.cloneTerragruntOptionsForDependencyOutput(0xc000c8[45](https://gitlab.com/<REMOVED>/-/jobs/8522364760#L45)80, {0xc00121c0c0, 0x5f})
	/home/circleci/project/config/dependency.go:693 +0x2f0
github.com/gruntwork-io/terragrunt/config.getTerragruntOutputJSON(0xc000c84580, {0xc00121c0c0, 0x5f})
	/home/circleci/project/config/dependency.go:743 +0x3a
github.com/gruntwork-io/terragrunt/config.getOutputJSONWithCaching(0xc000c84580, {0xc00121c0c0, 0x5f})
	/home/circleci/project/config/dependency.go:616 +0x345
github.com/gruntwork-io/terragrunt/config.getTerragruntOutput(0xc000c84580, {{0xc00033ef10, 0x8}, 0x0, {{{0x2e9dee0, 0xc000013151}}, {0x21dc0e0, 0xc0010a6cc0}}, 0x0, 0xc0003575e0, ...})
	/home/circleci/project/config/dependency.go:550 +0x90
github.com/gruntwork-io/terragrunt/config.getTerragruntOutputIfAppliedElseConfiguredDefault(0xc000c84580, {{0xc00033ef10, 0x8}, 0x0, {{{0x2e9dee0, 0xc000013151}}, {0x21dc0e0, 0xc0010a6cc0}}, 0x0, 0xc0003575e0, ...})
	/home/circleci/project/config/dependency.go:[47](https://gitlab.com/<REMOVED>/-/jobs/8522364760#L47)4 +0x1d8
github.com/gruntwork-io/terragrunt/config.(*Dependency).setRenderedOutputs(0xc000b998f0, 0x4b28fd?)
	/home/circleci/project/config/dependency.go:164 +0x19b
github.com/gruntwork-io/terragrunt/config.dependencyBlocksToCtyValue.func1()
	/home/circleci/project/config/dependency.go:414 +0x87
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/home/circleci/go/pkg/mod/golang.org/x/sync@v0.9.0/errgroup/errgroup.go:78 +0x[50](https://gitlab.com/<REMOVED>/-/jobs/8522364760#L50)
created by golang.org/x/sync/errgroup.(*Group).Go in goroutine 67
	/home/circleci/go/pkg/mod/golang.org/x/sync@v0.9.0/errgroup/errgroup.go:75 +0x96
```

- [ ] Screenshots

## Versions

- Terragrunt version: `0.69.1`
- Terraform version: `1.9.8`
- Environment details (Ubuntu 20.04, Windows 10, etc.):
  - AKSUbuntu-2204gen2containerd-202410.27.0

**Comment by yhakbar** [comment]
Hey @Tomasz-Kluczkowski ,

Thanks for reporting this. Taking a look.

**Comment by yhakbar** [comment]
Fix released in [v0.69.5](https://github.com/gruntwork-io/terragrunt/releases/tag/v0.69.5).
